servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/ConnectionPoolStrategyFactory.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.loadbalancer;
+
+import io.servicetalk.client.api.LoadBalancedConnection;
+
+/**
+ * The factory of {@link ConnectionPoolStrategy} instances.
+ * @param <C> the least specific connection type necessary for properly implementing the strategy.
+ * @see ConnectionPoolStrategies for available strategies.
+ */
+public interface ConnectionPoolStrategyFactory<C extends LoadBalancedConnection> {
+
+    /**
+     * Provide an instance of the {@link ConnectionPoolStrategy} to use with a {@link Host}.
+     * @param targetResource the resource name, used for logging purposes.
+     * @return an instance of the {@link ConnectionPoolStrategy} to use with a {@link Host}.
+     */
+    ConnectionPoolStrategy<C> buildStrategy(String targetResource);
```

Comment:
I'm ok to keep as-is if you prefer. Alternatively, consider just passing `targetResource` as the 3rd argument to `ConnectionPoolStrategy.select(...)`. That way we can reduce number of interfaces.